ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -60,10 +61,43 @@ type RayActorOptionSpec struct {
 	AcceleratorType   string                 `json:"accelerator_type,omitempty"`
 }
 
-// ServeDeploymentStatuses defines the current states of all Serve Deployments.
+type AppStatusV1 struct {
+	Status  string `json:"status,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+type ServeDeploymentStatusV1 struct {
+	Name    string `json:"name,omitempty"`
+	Status  string `json:"status,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+// (V1) ServeDeploymentStatuses defines the current states of all Serve Deployments.
 type ServeDeploymentStatuses struct {
-	ApplicationStatus  rayv1alpha1.AppStatus               `json:"app_status,omitempty"`
-	DeploymentStatuses []rayv1alpha1.ServeDeploymentStatus `json:"deployment_statuses,omitempty"`
+	ApplicationStatus  AppStatusV1               `json:"app_status,omitempty"`
+	DeploymentStatuses []ServeDeploymentStatusV1 `json:"deployment_statuses,omitempty"`
+}
+
+type ServeDeploymentDetails struct {
+	Name    string `json:"name"`
+	Status  string `json:"status"`
+	Message string `json:"message"`
+}
+
+type ServeApplicationDetails struct {
+	Name                    string                            `json:"name"`
+	RoutePrefix             string                            `json:"route_prefix"`
+	DocsPath                string                            `json:"docs_path"`
+	Status                  string                            `json:"status"`
+	Message                 string                            `json:"message"`
+	LastDeployedTimeSeconds float32                           `json:"last_deployed_time_s"`
+	Deployments             map[string]ServeDeploymentDetails `json:"deployments"`
+}
+
+//
+type ServeDetails struct {
```

Comment:
Seems like you've identified two things:
1. The `RayServiceStatus` also has a field called `Applications`. For conciseness I called the fields in `RayServiceStatus`: `Applications` and `Deployments` instead of `ApplicationStatuses` and `DeploymentStatuses`. Would it be clearer if I changed it back?
2. You didn't realize `serveDetails` is a map. For this PR, a map would definitely work. However the json returned in the response from the rest api contains more info than just applications, so if we wanted to fetch that information we would need to add the struct back in the future. Would you prefer we use the struct only when necessary?